### PR TITLE
Sliders: Add Autoplay and Autoplay Pause on Hover settings

### PR DESCRIPTION
**Pull request by AlexGStapleton**
This PR resolves https://github.com/siteorigin/so-widgets-bundle/issues/1101 https://github.com/siteorigin/so-widgets-bundle/issues/982 https://github.com/siteorigin/so-widgets-bundle/issues/920

### CODE REVIEW

**Comment by AlexGStapleton** [comment]
The last PR will prevent the Animation Speed setting from being ever hidden as it affects both the autoplay and non-autoplay settings.

**Comment by Misplon** [comment]
Thanks. Looks like this PR is currently removing Timeout when Autoplay isn't enabled. Timeout is currently a setting available in develop.

**Comment by AlexGStapleton** [comment]
That was done due to the request made in #1101

> Can we make the **timeout** and animation settings appear when this checkbox is enabled and **hide when disabled**?

Animation speed is used regardless of autoplay, but timeout only affects the slider if autoplay is enabled so it makes sense to hide it if autoplay is disabled. Regardless, if you would prefer that timeout is always visible let me know and I'll remove the state emitter.

**Comment by Misplon** [comment]
Thanks. If the timeout setting is only used when autoplay is enabled then we good.

File: base/inc/widgets/base-slider.class.php [hunk 1]
```diff
@@ -162,6 +179,7 @@ function slider_settings( $controls ){
 			'pagination'               => true,
 			'speed'                    => empty( $controls['speed'] ) ? 1 : $controls['speed'],
 			'timeout'                  => $controls['timeout'],
+			'paused'                 => empty( $controls['autoplay'] ) ?: false,
```
Comment (Misplon) [inline base/inc/widgets/base-slider.class.php:182]: There is perhaps an alignment issue, a space missing.

**Comment by Misplon** [comment]
Thanks.

In the Slider widget, try setting timeout to 1000 and enable pause on hover. Is pause on hover working for you, it isn't currently for me.

**Comment by Misplon** [comment]
Do you want to add a slide animation or some animation when hiding and showing the setting?

**Comment by AlexGStapleton** [comment]
> In the Slider widget, try setting timeout to 1000 and enable pause on hover. Is pause on hover working for you, it isn't currently for me. 

I just created a build to check is if this is working as expected and it is for. Here's a video of it working for me:

https://drive.google.com/uc?id=1GMTXm_3v8mdHJba0iStSNluevI2NEZK5

(please excuse the mouse placement - that's due to display scaling)

[Here's the build I tried](https://drive.google.com/uc?id=1GiHFkKws-hV1A-1xC8c-Pm6Q1ZQHMpSl).

> Do you want to add a slide animation or some animation when hiding and showing the setting?

Sure. That sort of change should however be in a separate PR as that change will affect all state emitters.

**Comment by Misplon** [comment]
Please, log the state emitter transition task and self-assign. Thanks.

Thanks for the build, all looks good in there.

**Pull request merged**